VideoApi/VideoApi/Controllers/ConsultationController.cs

```diff
@@ -241,14 +150,76 @@ public async Task<IActionResult> StartPrivateConsultationWithEndpointAsync(Endpo
+
+            return Ok();
+        }
 
-            return Accepted();
+        [HttpPost("lockroom")]
+        [OpenApiOperation("LockRoom")]
+        [ProducesResponseType((int)HttpStatusCode.OK)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
+        [ProducesResponseType((int)HttpStatusCode.Unauthorized)]
+        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+        public async Task<IActionResult> LockRoomRequestAsync(LockRoomRequest request)
+        {
+            try
+            {
+                var lockRoomCommand = new LockRoomCommand(request.ConferenceId, request.RoomLabel, request.Lock);
+                await _commandHandler.Handle(lockRoomCommand);
+                return Ok();
+            }
+            catch (RoomNotFoundException ex)
+            {
+                _logger.LogError(ex, "Room doest not exist in conference {conferenceId} with label {label}", request.ConferenceId, request.RoomLabel);
+                return NotFound("Room does not exist");
+            }
+        }
+
+        [HttpPost("createconsultation")]
+        [OpenApiOperation("CreatePrivateConsultation")]
+        [ProducesResponseType(typeof(RoomResponse), (int)HttpStatusCode.OK)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
+        [ProducesResponseType((int)HttpStatusCode.Unauthorized)]
+        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+        public async Task<IActionResult> StartNewConsultationRequestAsync(StartConsultationRequest request)
+        {
+            try
+            {
+                var room = await _consultationService.CreateNewConsultationRoomAsync(request.ConferenceId);
+                await _consultationService.JoinConsultationRoomAsync(request.ConferenceId, request.RequestedBy, room.Label);
+
+                var response = RoomToDetailsResponseMapper.MapRoomToResponse(room);
+                return Ok(response);
+            }
+            catch (ConferenceNotFoundException ex)
+            {
+                _logger.LogError(ex,
+                    "Cannot create consultation for conference: {conferenceId} as the conference does not exist",
+                    request.ConferenceId);
+                return NotFound("Conference does not exist");
+            }
+            catch (ParticipantNotFoundException ex)
+            {
+                _logger.LogError(ex,
+                    "Cannot create consultation with participant: {participantId} as the participant does not exist",
+                    request.RequestedBy);
+                return NotFound("Participant doesn't exist");
+            }
+            catch (KinlyApiException ex)
+            {
+                _logger.LogError(ex,
+                    "Unable to create a consultation room for ConferenceId: {conferenceId}",
+                    request.ConferenceId);
+                return BadRequest("Consultation room creation failed");
+            }
         }
 
         [HttpPost("start")]
```

Comment:
@levibutler  what is the reason for changing these? Start and End were supposed to be only for JOH consultations as they follow a different logic to participant consultations.  Implementation should have just removed/replaced the existing participant consultation logic and left the JOH endpoints as is.